d2/src/test/java/com/linkedin/d2/balancer/strategies/relative/StateUpdaterTest.java

```diff
@@ -524,6 +523,34 @@ Object[][] getTrackerClients()
         };
   }
 
+  @Test
+  public void testStreamError()
+  {
+    Map<ErrorType, Integer> errorTypeCounts = new HashMap<>();
+    errorTypeCounts.put(ErrorType.STREAM_ERROR, 11);
+    List<TrackerClient> trackerClients = TrackerClientMockHelper.mockTrackerClients(2,
+        Arrays.asList(20, 20), Arrays.asList(10, 10), Arrays.asList(200L, 500L), Arrays.asList(100L, 200L), Arrays.asList(0, 11),
+        false, Arrays.asList(false, false), Arrays.asList(Collections.emptyMap(), errorTypeCounts));
+
+    PartitionState state = new PartitionStateTestDataBuilder()
+        .setClusterGenerationId(DEFAULT_CLUSTER_GENERATION_ID)
+        .setTrackerClientStateMap(trackerClients,
+            Arrays.asList(StateUpdater.MAX_HEALTH_SCORE, StateUpdater.MAX_HEALTH_SCORE),
+            Arrays.asList(TrackerClientState.HealthState.HEALTHY, TrackerClientState.HealthState.HEALTHY),
+            Arrays.asList(30, 30))
+        .build();
+
+    ConcurrentMap<Integer, PartitionState> partitionLoadBalancerStateMap = new ConcurrentHashMap<>();
+    partitionLoadBalancerStateMap.put(DEFAULT_PARTITION_ID, state);
+    setup(new D2RelativeStrategyProperties().setHighErrorRate(0.5), partitionLoadBalancerStateMap);
```

Comment:
The test utilizes Relative strategy, but the `StateUpdater` code is shared b/w relative and degrader, so it should be okay.